### PR TITLE
🙋‍♀️ Multiple question support

### LINKED ISSUE #26: Submission Page

### LINKED ISSUE #53: Move default problem to firebase
At the moment the default problem is stored in-code. This should be moved to firebase.

Scoping should be done for having multiple problems via firebase.

### LINKED ISSUE #59: Fix highlights + usedParsonsIds fields for new users

### LINKED ISSUE #60: Multiple question support

### LINKED ISSUE #61: Assignment Support

### PR DESCRIPTION
**Pull request by kand198**
## Description

* Refactored code related to Firestore to allow for multiple questions to exist. Multiple questions can now be handled gracefully.
* Added support for assignments
* Added assignment selection page
* Added flow between questions in assignment
* Added assignment submission page incl. feedback form.

Closes #61,
Closes #60,
Closes #59, 
Closes #53,
Closes #26

## How has this been tested?

Tested locally. Probably introduced bugs. Should be thoroughly tested.

## Checklist

- [x] I have performed a self-review of my own code
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass

### CODE REVIEW

File: src/components/AssignmentFeedbackForm/AssignmentFeedbackForm.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        <br />
+        Important: you must submit to coderunner to get your marks for this
+        course!
+      </Text>
+      <InputWrapper label='How easy was it to use this tool?' required>
+        <Slider
+          label={(val) => easyMarks?.find((mark) => mark.value === val)?.label}
+          step={5}
+          marks={easyMarks}
+          // eslint-disable-next-line react/jsx-props-no-spreading
```
Comment (uoa-ypec413) [inline src/components/AssignmentFeedbackForm/AssignmentFeedbackForm.tsx:70]: Could you just disable this in the eslint config file rules?
`'react/jsx-props-no-spreading': 'off'`

**Review by uoa-ypec413 — CHANGES_REQUESTED** [review]
I've tested and the functionality is great. I just have a few changes I'd like to see:

1. Could you add a real question to the second question? It would be nice to confirm that each stage is working correctly when we skip to the next question.

2. Could you remove the forward + back buttons in the navigation bar? They're unused, easy to remove in this PR.

3. When I go to submit the last question, the tool still thinks I'm on question 1/2.

4. When the student has submitted their code, could we make the assignment card green with a tick on the Assignment entry page?

5. Could we add a copy button to the last page?

6. UI Changes:
- Try center our header, and adjust it to look like the Figma
- For the feedback form, could you add what is in the Figma?

If you want to add these to a separate ticket that's fine.

File: src/components/AssignmentFeedbackForm/AssignmentFeedbackForm.tsx [hunk 2]
```diff
@@ -54,59 +66,106 @@ const AssignmentFeedbackForm = () => {
...
+            size='lg'
+            step={5}
+            marks={marks}
+            classNames={{
+              bar: 'bg-blue-600',
+              markFilled: 'border-blue-600',
+              thumb: 'border-blue-600',
+              markLabel: 'font-bold text-gray-800',
+            }}
+            // eslint-disable-next-line react/jsx-props-no-spreading
```
Comment (uoa-ypec413) [inline src/components/AssignmentFeedbackForm/AssignmentFeedbackForm.tsx:103]: You've already turned this off in the rules so I don't believe this is necessary anymore?

**Review by uoa-ypec413 — CHANGES_REQUESTED** [review]
Aside from the comment below as well as the feedback form spacing change we discussed, LGTM. Thanks for making the requested changes.
🕷️

File: src/components/AssignmentFeedbackForm/AssignmentFeedbackForm.tsx (hunk 2 above)
Comment (kand198) [reply src/components/AssignmentFeedbackForm/AssignmentFeedbackForm.tsx:103]: dang i forgot sorry

File: src/components/AssignmentFeedbackForm/AssignmentFeedbackForm.tsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
```
Comment (uoa-ypec413) [inline src/components/AssignmentFeedbackForm/AssignmentFeedbackForm.tsx:103]: There's still some here pls just ctrl-f it

**Review by uoa-ypec413 — APPROVED** [review]
😍😍😍😍😍

**Pull request merged**